docs/af-fundamentals-tutorial/modules/deadlines/pages/implementing.adoc

```diff
@@ -0,0 +1,77 @@
+= Implementing Deadlines
+:navtitle: Add a handler
+
+As the `SimpleDeadlineManager` is included in the messaging module of Axon framework, and uses Java's `ScheduledExecutorService` no additional dependencies are needed to use it.
+The `SimpleDeadlineManager` can be added both as an Axon Configuration component, or as a Spring Bean.
+For this howto we will configure it as a Spring Bean.
+
+[source,java]
+----
+@Bean
+public DeadlineManager deadlineManager(
+        org.axonframework.config.Configuration configuration
+) {
+    var provider = new ConfigurationScopeAwareProvider(configuration);
+    return SimpleDeadlineManager
+            .builder()
+            .scopeAwareProvider(provider)
+            .build();
+}
+----
+
+There are some additional things that could be set.
+All deadline managers need at least the scope-aware provider.
+
+[source,java]
+----
+@Aggregate
+public class ExpiringGiftCard {
+
+    @AggregateIdentifier
+    private String giftCardId;
+    private boolean expired;
+    // some other state left out for brevity
+
+    private static final String EXPIRED_GIFT_CARD = "EXPIRED_GIFT_CARD";
+
+    @CommandHandler
+    @CreationPolicy(AggregateCreationPolicy.CREATE_IF_MISSING)
+    public void handle(IssueExpiringCardCommand command, DeadlineManager deadlineManager) { // <1>
+        //check validity
+        apply(new ExpiringCardIssuedEvent(command.id(), command.daysValid(), command.amount()));
+        Instant trigger = Instant.now().plus(command.daysValid(), ChronoUnit.DAYS).truncatedTo(ChronoUnit.DAYS);
+        deadlineManager.schedule(trigger, EXPIRED_GIFT_CARD); // <2>
+    }
+
+    @DeadlineHandler(deadlineName = EXPIRED_GIFT_CARD) // <3>
+    public void on() {
+        if (!expired){ //<4>
+            apply(new ExpiringCardExpiredEvent(giftCardId));
+        }
+    }
+
+    @EventSourcingHandler
+    public void on(ExpiringCardIssuedEvent event) {
+        giftCardId = event.id();
+        expired = false;
+    }
+
+    @EventSourcingHandler // <5>
+    public void on(ExpiringCardExpiredEvent event) {
+        expired = true;
+    }
+
+    // More handlers are needed to make it useful, left out for brevity.
+
+    public ExpiringGiftCard() {
+        // Required by Axon to construct an empty instance to initiate Event Sourcing.
+    }
+}
+----
+
+<1> Adding the deadline manager to the method makes it easily testable compared to other ways of accessing it from the aggregate.
```

Comment:
I don't think there *is* any other way, right? 
It would be nice to explain here that Axon Framework injects this from the Spring Context or Configuration, so we can use it

Reply:
You can `Autowire` them on the aggregate or saga, but in that case they are not testable with fixtures. Or use any other dependency injection, but it would be compatible with the test fixture. I'll ad we can inject from either the axon configuration or spring context.